### PR TITLE
feat(auth): allow backend credential to update publication #NP-47837

### PR DESCRIPTION
**Pull request by torbjokv**
dependent on this https://github.com/BIBSYSDEV/nva-identity-service/pull/377

### CODE REVIEW

File: publication-commons/src/main/java/no/unit/nva/publication/RequestUtil.java [hunk 1]
```diff
@@ -84,7 +85,15 @@ private static UserInstance createExternalUserInstance(RequestInfo requestInfo,
             client.getCristinUrgUri()
         );
 
-        return UserInstance.createExternalUser(resourceOwner, client.getCustomerUri());
+        UserInstance user;
```
Comment (monaullah) [inline publication-commons/src/main/java/no/unit/nva/publication/RequestUtil.java:88]:
```
return requestInfo.clientIsInternalBackend()
                   ? UserInstance.createBackendUser(resourceOwner, client.getCustomerUri())
                   : UserInstance.createExternalUser(resourceOwner, client.getCustomerUri());
```

File: publication-commons/src/main/java/no/unit/nva/publication/RequestUtil.java [hunk 2]
```diff
@@ -93,14 +102,14 @@ private static UserInstance createInternalUserInstance(RequestInfo requestInfo)
...
     }
 
     public static UserInstance createUserInstanceFromRequest(RequestInfo requestInfo,
                                                              IdentityServiceClient identityServiceClient)
         throws UnauthorizedException {
         try {
-            return requestInfo.clientIsThirdParty()
+            return requestInfo.clientIsThirdParty() || requestInfo.clientIsInternalBackend()
                        ? createExternalUserInstance(requestInfo, identityServiceClient)
                        : createInternalUserInstance(requestInfo);
```
Comment (truhacevkir) [inline publication-commons/src/main/java/no/unit/nva/publication/RequestUtil.java:114]: Strange naming clientIsInternalBacked -> createExternalUser.. :)

File: publication-commons/src/main/java/no/unit/nva/publication/permission/strategy/grant/BackendClientStrategy.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
+package no.unit.nva.publication.permission.strategy.grant;
+
+import no.unit.nva.model.Publication;
+import no.unit.nva.model.PublicationOperation;
+import no.unit.nva.publication.model.business.UserInstance;
+import no.unit.nva.publication.service.impl.ResourceService;
+
+public class BackendClientStrategy extends GrantPermissionStrategy {
```
Comment (truhacevkir) [inline publication-commons/src/main/java/no/unit/nva/publication/permission/strategy/grant/BackendClientStrategy.java:8]: Nice

**Review by truhacevkir — APPROVED** [review]

File: publication-commons/src/main/java/no/unit/nva/publication/RequestUtil.java (hunk 1 above)
Comment (torbjokv) [reply publication-commons/src/main/java/no/unit/nva/publication/RequestUtil.java]: thanks!

File: publication-commons/src/main/java/no/unit/nva/publication/RequestUtil.java (hunk 2 above)
Comment (torbjokv) [reply publication-commons/src/main/java/no/unit/nva/publication/RequestUtil.java]: I agree, not sure if I want to change requestInfo.clientIsInternalBackend today, but I will change the other functions to differentiate

**Pull request merged**